### PR TITLE
disable price calculation for free price

### LINKED ISSUE #1168: Unnecessary "Calculating price incl. fees"
This text and loading flow should only appear on pool assets, as there is nothing to calculate for fixed or free pricing. Fixed/Free assets should get the action button right away.

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #1168.

### CODE REVIEW

**Review by claudiaHash — APPROVED** [review]
LGTM

File: src/components/Asset/AssetActions/Download.tsx [hunk 1]
```diff
@@ -73,7 +73,11 @@ export default function Download({
     setValidOrderTx(asset?.accessDetails?.validOrderTx)
     // get full price and fees
     async function init() {
-      if (asset?.accessDetails?.addressOrId === ZERO_ADDRESS) return
+      if (
+        asset?.accessDetails?.addressOrId === ZERO_ADDRESS ||
```
Comment (EnzoVezzaro) [inline src/components/Asset/AssetActions/Download.tsx:77]: I couldn't verify the need for this. In which case does the address equal to ZERO_ADDRESS? is this really need it?

Comment (KatunaNorbert) [reply src/components/Asset/AssetActions/Download.tsx:77]: I don't know why that check is in there, that's why I didn't touch it.

Comment (mihaisc) [reply src/components/Asset/AssetActions/Download.tsx:77]: that check is for the publish preview

**Review by EnzoVezzaro — APPROVED** [review]

File: src/components/Asset/AssetActions/Download.tsx [hunk 2]
```diff
@@ -73,7 +73,11 @@ export default function Download({
...
+        asset?.accessDetails.type === 'free'
```
Comment (kremalicious) [inline src/components/Asset/AssetActions/Download.tsx:78]: if there's `asset?.accessDetails?` everywhere else, the `?` should be set here too

**Review by EnzoVezzaro — APPROVED** [review]

**Pull request merged**